[iOS] iOS AV1 HW decoder requires to get the OBU header in its av1C

youennf opened the pull request:
#### 208a1c80738f71e950ea77254d58aeb7b15df747
<pre>
[iOS] iOS AV1 HW decoder requires to get the OBU header in its av1C
<a href="https://bugs.webkit.org/show_bug.cgi?id=269906">https://bugs.webkit.org/show_bug.cgi?id=269906</a>
<a href="https://rdar.apple.com/121924090">rdar://121924090</a>

Reviewed by Jean-Yves Avenard.

iOS AV1 decoder needs the header OBU in its av1C to properly initialize itself.
We update getSequenceHeaderOBU to return bothe the full OBU and the obu data.
We then update computeAV1InputFormat to append the full OBU to the av1C data.

Manually tested using <a href="https://webrtc.github.io/samples/src/content/peerconnection/change-codecs/">https://webrtc.github.io/samples/src/content/peerconnection/change-codecs/</a> on iPhone 15 Pro.

* Source/WebCore/platform/video-codecs/cocoa/RTCVideoDecoderVTBAV1.mm:
(getSequenceHeaderOBU):
(computeAV1InputFormat):

Canonical link: <a href="https://commits.webkit.org/275172@main">https://commits.webkit.org/275172@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/04ac3e4279ea08ebfb1a39a37e1c49fd27beefbc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41076 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20089 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43454 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43637 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37168 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43383 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23113 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17420 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/34006 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41650 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17012 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35377 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14639 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14763 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36376 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44950 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37267 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36689 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40438 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15891 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13027 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38815 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17510 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9225 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17562 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17154 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/platform/video-codecs/cocoa/RTCVideoDecoderVTBAV1.mm:
```diff
@@ -263,14 +267,21 @@ static size_t readULEBSize(std::span<const uint8_t> data, size_t& index)
-    uint8_t header[VPCodecConfigurationContentsSize];
+    size_t cfDataSize = VPCodecConfigurationContentsSize + fullOBUHeader.size();
+    auto cfData = adoptCF(CFDataCreateMutable(kCFAllocatorDefault, cfDataSize));
+    CFDataIncreaseLength(cfData.get(), cfDataSize);
```
is this required? isn't the size already cfDataSize?

jyavenard reviewed (approved):
comment:
s/to return bothe/to return both/

youennf committed:
[iOS] iOS AV1 HW decoder requires to get the OBU header in its av1C
https://bugs.webkit.org/show_bug.cgi?id=269906
rdar://121924090

Reviewed by Jean-Yves Avenard.

iOS AV1 decoder needs the header OBU in its av1C to properly initialize itself.
We update getSequenceHeaderOBU to return bothe the full OBU and the obu data.
We then update computeAV1InputFormat to append the full OBU to the av1C data.

Manually tested using https://webrtc.github.io/samples/src/content/peerconnection/change-codecs/ on iPhone 15 Pro.

* Source/WebCore/platform/video-codecs/cocoa/RTCVideoDecoderVTBAV1.mm:
(getSequenceHeaderOBU):
(computeAV1InputFormat):

Canonical link: https://commits.webkit.org/275172@main
